Update /user

MarkGibbons opened the pull request:
### Description

Update the global /user endpoints.  Document the api version 1 requests and responses.

### Definition of Done

### Issues Resolved

[List any existing issues this PR resolves, or any Discourse or
StackOverflow discussion that's relevant]

### Check List

- [ ] Spell Check
- [ ] Local build
- [ ] Examine the local build
- [ ] All tests pass

Commit:
full_name does not appear to be a field for users

Commit:
Update the /user and /user/NAME endpoints

The existing doc was sort of for api version 0.  This update creates the
doc for api version 1.

Signed-off-by: markgibbons <mark.gibbons@nordstrom.com>

Commit:
Editing

Signed-off-by: IanMadd <imaddaus@chef.io>

IanMadd commented on content/api_chef_server.md:
```diff
@@ -931,26 +938,33 @@ where:
 -   `username` must begin with a lower-case letter or digit, may only
     contain lower-case letters, digits, hyphens, and underscores. For
     example: `chef`.
+    `username` is required to be present and have a valid value. A valid
```
Should "name" on line 924 be changed to "username" or should this be "name"?

MarkGibbons replied:
yes 924 should be username.  Thank you for catching that.  Where I have chefserver in the URIs it would be more consistent to use chef.example.  I'll fix and repost.

MarkGibbons replied:
reposted

IanMadd replied:
Could you add your DCO signoff too. Thanks!

MarkGibbons replied:
Done.

Commit:
Merge branch 'master' into update_user

Commit:
Fix the username field name in /users POST

Update the chef server name in some URIs to be more consistent across
the documentation.

One URI in associations was just wrong, it was missing the server name
completely.

Signed-off-by: markgibbons <mark.gibbons@nordstrom.com>

IanMadd committed:
Merge branch 'master' into update_user

IanMadd committed:
Merge branch 'master' into update_user

IanMadd reviewed (approved):
![tenor-251404126](https://user-images.githubusercontent.com/8138169/76342124-44bf1000-62bb-11ea-80b4-203dcaddd57f.gif)
